boa/src/builtins/function/arguments.rs

```diff
@@ -160,87 +161,83 @@ impl Arguments {
+                if property_index >= len {
+                    break 'outer;
                 }
+                let binding_index = bindings.len() + 1;
+                let entry = bindings
+                    .entry(name)
+                    .or_insert((binding_index, property_index));
+                entry.1 = property_index;
+                property_index += 1;
             }
         }
+        for (binding_index, property_index) in bindings.values() {
+            // 1. Let g be MakeArgGetter(name, env).
+            // https://tc39.es/ecma262/#sec-makearggetter
+            let g = {
+                // 2. Let getter be ! CreateBuiltinFunction(getterClosure, 0, "", « »).
+                // 3. NOTE: getter is never directly accessible to ECMAScript code.
+                // 4. Return getter.
+                FunctionBuilder::closure_with_captures(
+                    context,
+                    // 1. Let getterClosure be a new Abstract Closure with no parameters that captures
+                    // name and env and performs the following steps when called:
+                    |_, _, captures, _| Ok(captures.0.get(captures.1)),
+                    (env.clone(), *binding_index),
+                )
+                .length(0)
+                .name("")
+                .build()
+            };
+            // 2. Let p be MakeArgSetter(name, env).
+            // https://tc39.es/ecma262/#sec-makeargsetter
+            let p = {
+                // 2. Let setter be ! CreateBuiltinFunction(setterClosure, 1, "", « »).
+                // 3. NOTE: setter is never directly accessible to ECMAScript code.
+                // 4. Return setter.
+                FunctionBuilder::closure_with_captures(
+                    context,
+                    // 1. Let setterClosure be a new Abstract Closure with parameters (value) that captures
+                    // name and env and performs the following steps when called:
+                    |_, args, captures, _| {
+                        let value = args.get(0).cloned().unwrap_or_default();
+                        captures.0.set(captures.1, value);
+                        Ok(JsValue::undefined())
+                    },
+                    (env.clone(), *binding_index),
+                )
+                .length(1)
+                .name("")
+                .build()
+            };
+
+            // 3. Perform map.[[DefineOwnProperty]](! ToString(𝔽(index)), PropertyDescriptor {
+            // [[Set]]: p, [[Get]]: g, [[Enumerable]]: false, [[Configurable]]: true }).
+            map.__define_own_property__(
+                PropertyKey::from(*property_index),
+                PropertyDescriptor::builder()
+                    .set(p)
+                    .get(g)
+                    .enumerable(false)
+                    .configurable(true)
+                    .build(),
+                context,
+            )
+            .expect("[[DefineOwnProperty]] must not fail per the spec");
```

Comment:
If it must not fail, can it not return a `Result`? (maybe it's out of the scope of this PR)

Reply:
Probably the message I left was not descriptive enough. It should be changed to "Defining new own properties for a new ordinary object cannot fail" or something similar.

Reply:
Changed all of the `expect` messages in that file to that wording, as they all boil down to that.